source/developer/interactive-messages.rst

```diff
@@ -0,0 +1,329 @@
+.. _interactive-messages:
+
+Interactive Messages
+============================
+
+Mattermost supports interactive messages for `incoming <https://docs.mattermost.com/developer/webhooks-incoming.html>`_ and `outgoing webhooks <https://docs.mattermost.com/developer/webhooks-outgoing.html>`_, `custom slash commands <https://docs.mattermost.com/developer/slash-commands.html>`_ and `plugins <https://docs.mattermost.com/administration/plugins.html>`_ via actions. They help make your integrations richer by completing common tasks inside Mattermost conversations, increasing user engagement and productivity.
+
+Use interactive messages to simplify complex workflows by allowing users to take quick actions directly through your integration post. For example, they enable your integration to:
+
+- mark a task complete in your project management tracker
+- conduct a customer survey or a poll
+- initiate a command to merge a branch into a release
+
+To try it out, you can use this `matterpoll plugin <https://github.com/matterpoll/matterpoll>`_ to add polling to Mattermost channels via a ``/poll`` slash command.
+
+.. image:: ../../source/images/matterpoll.png
+
+.. toctree::
+  :maxdepth: 2
+
+Message Buttons
+----------------
+
+Add message buttons as ``actions`` in your integration `message attachments <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+.. image:: ../../source/images/poll.gif
+
+The following payload gives an example that uses message buttons.
+
+.. code-block:: text
+
+  {
+    "attachments": [
+      {
+        "pretext": "This is the attachment pretext.",
+        "text": "This is the attachment text.",
+        "actions": [
+          {
+            "name": "Ephemeral Message",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "do_something_ephemeral"
+              }
+            }
+          }, {
+            "name": "Update",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "do_something_update"
+              }
+            }
+          }
+        ]
+      }
+    ]
+  }
+
+The integration can respond with an update to the original post, or with an ephemeral message:
+
+.. code-block:: text
+
+  {
+    "update": {
+      "message": "Updated!"
+    },
+    "ephemeral_text": "You updated the post!"
+  }
+
+.. image:: ../../source/images/interactive_message.gif
+
+Message Menus
+----------------
+
+Similar to buttons, add message menus as ``actions`` in your integration `message attachments <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+.. image:: ../../source/images/message_menus.png
```

Comment:
Image doesn't render, looks like it might be `message-menus.png`? 

Also it looks like it may be blurry compared to some of the other images on the page, so we may want to update the image to be more clear. 

<img width="709" alt="screenshot 2018-10-10 17 22 58" src="https://user-images.githubusercontent.com/9354841/46773294-793b3080-ccb1-11e8-85e2-8f8e23d8af26.png">